### PR TITLE
docs: update examples page

### PR DESCRIPTION
**Pull request by MrPowers**
This PR introduces a new approach for the [Apache Spark Examples page](https://spark.apache.org/examples.html).  Here are some areas of improvement:

* The current guide has "Additional Examples" as the page h1, not "Apache Spark Examples"
* We're not showing the underlying data, so the code snippets are hard to follow
* Python snippets aren't PEP 8 formatted
* Aren't showing SQL syntax
* The ML example is using MLlib
* The average time on page is pretty low, so users probably aren't too engaged:

Screenshot 2023-12-15 at 9 53 06 AM

This revised doc provides a gentle introduction that users might find more approachable.  Interested in thoughts from the community.

It's a draft PR to get feedback and because the code blocks still need to be added for all the languages.

### CODE REVIEW

**Comment by zhengruifeng** [comment]
I think you can generate the html changes by following [Generating the website HTML](https://github.com/apache/spark-website)

**Review by zhengruifeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by allisonwang-db — APPROVED** [review]
This is great! Thanks for updating it.
